src/components/Select/Select.tsx

```diff
@@ -577,7 +578,11 @@ export const Select: FC<SelectProps> = React.forwardRef(
             disabled={mergedDisabled}
             key={`select-pill-${index}`}
             label={opt.text}
-            onClose={() => toggleOption(opt)}
+            onClose={(event) => {
+              if (contains(document.activeElement, event.target as Node)) {
```

Comment:
whats the idea here? why does it toggle option on close at all? seems like onclick/onkeydown should be solely responsible for toggling an option

Reply:
I only just now realized that this is the pill onClose and not the dropdown on close... is there a possibly a propagation issue in play here?

Reply:
No. I’ve debugged the propagation part. event propagation is not affecting the onClose behavior for the pill component.

Reply:
Do you have an understanding of how this is getting triggered?